src/interfaces/MutatorFunction.ts

```diff
@@ -1,4 +1,4 @@
 import ActionMessage from './ActionMessage';
 
-type MutatorFunction<T extends ActionMessage> = (actionMessage: T) => void;
+type MutatorFunction<T extends ActionMessage, U> = void extends U ? (actionMessage: T) => U : never;
```

Comment:
>T extends ActionMessage, U [](start = 21, length = 26)

Can you rename these type params to make clear what they are?  Probably `TAction` and `TReturn`.